helpers.go

```diff
@@ -139,3 +141,32 @@ func decodeSlice(elemType reflect.Type, slice reflect.Value, input map[string]in
 
 	return nil
 }
+
+// customDecodeHook supports decoding to time
+func customDecodeHook(from reflect.Type, to reflect.Type, data interface{}) (interface{}, error) {
+
+	// Decode time.Time
+	if to == reflect.TypeOf(time.Time{}) {
+		var timeInt int64
+		var err error
+
+		// From int
+		if from == reflect.TypeOf(0) {
```

Comment:
Use a switch here as it cant be both.
Also not sure the compiler will optimise out the TypeOf so these could be internal vars.

Alternatively comparing against the Kind() would also work, see StringToTimeDurationHookFunc for an example.